### PR TITLE
add RGBA Encode support

### PR DESCRIPTION
**Pull request by codeskyblue**

### CODE REVIEW

**Comment by codeskyblue** [comment]
ref <https://github.com/pixiv/go-libjpeg/issues/19>

**Comment by codeskyblue** [comment]
Really wish to merge soon.

File: jpeg/compress.go [hunk 1]
```diff
@@ -175,6 +188,24 @@ func encodeYCbCr(cinfo *C.struct_jpeg_compress_struct, src *image.YCbCr, p *Enco
 	return
 }
 
+// encode image.RGBA
+func encodeRGBA(cinfo *C.struct_jpeg_compress_struct, src *image.RGBA, p *EncoderOptions) (err error) {
+	// Set up compression parameters
+	cinfo.image_width = C.JDIMENSION(src.Bounds().Dx())
+	cinfo.image_height = C.JDIMENSION(src.Bounds().Dy())
+	cinfo.input_components = 4
+	cinfo.in_color_space = C.JCS_EXT_RGBA
```
Comment (harukasan) [inline jpeg/compress.go]: It needs to check JCS_EXT_RGBA support.
(see: https://github.com/pixiv/go-libjpeg/blob/master/jpeg/decompress.go#L299)

File: jpeg/compress.go [hunk 2]
```diff
@@ -44,6 +44,17 @@ static void encode_gray(j_compress_ptr cinfo, JSAMPROW pix, int stride) {
 	}
 }
 
+static void encode_rgba(j_compress_ptr cinfo, unsigned char pix[], int stride) {
```
Comment (harukasan) [inline jpeg/compress.go]: use `JSAMPROW` type for pix

File: jpeg/compress.go [hunk 3]
```diff
@@ -44,6 +44,17 @@ static void encode_gray(j_compress_ptr cinfo, JSAMPROW pix, int stride) {
...
+	JSAMPROW row_pointer[1];
+
+	int v;
+	for (v = 0; v < cinfo->image_height; v++) {
+		row_pointer[0] = &pix[v * stride];
+		jpeg_write_scanlines(cinfo, row_pointer, 1);
```
Comment (harukasan) [inline jpeg/compress.go]: use `v += jpeg_write_scanlines(cinfo, row_pointer, 1);` instead of increment in for statement.

**Comment by codeskyblue** [comment]
sorry not very familiar with cgo. maybe you can change directly

**Comment by harukasan** [comment]
@codeskyblue I'll continue working in #33. Thank you for your contributes!

**Pull request closed without merging**